src/scheduler/simulate.c

```diff
@@ -860,7 +860,7 @@ create_events(server_info *sinfo)
 	qsort(all, count_array((void **)all), sizeof(resource_resv *), cmp_events);
 
 	for (i = 0; all[i] != NULL && is_timed(all[i]); i++) {
-		if (all[i]->is_resv && all[i]->resv &&
+		if (all[i]->is_resv && all[i]->resv != NULL &&
```

Comment:
Why did you add the != NULL?  And then to only one of them instead of the first one too?

Reply:
It's the scheduler's style.  The scheduler will check if (ptr != NULL) instead of if (ptr).  This was introduced in a previous commit.  I asked Arun to fix it.

Reply:
Thanks for the response @bhroam.  I get it now.
Looks good to me.